src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/LegalRepOrganisationFormatter.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.*;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.AddressUk;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ref.OrganisationEntityResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.ProfessionalOrganisationRetriever;
+
+
+@Slf4j
+@Service
+public class LegalRepOrganisationFormatter implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private ProfessionalOrganisationRetriever professionalOrganisationRetriever;
+
+    public LegalRepOrganisationFormatter(ProfessionalOrganisationRetriever professionalOrganisationRetriever) {
+        this.professionalOrganisationRetriever = professionalOrganisationRetriever;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.START_APPEAL;
+
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage,
+                                                        Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
```

Comment:
@bradsorour can you add OrganisationPolicy set up to this handler? I think that way we can call PRD endpoint (`retrieve()` method) one time and use it for two goals.

Reply:
Yes, indeed. I've moved OrganisationPolicy into this handler.